kayenta-sql/src/main/java/com/netflix/kayenta/sql/migration/StorageDataMigrator.java

```diff
@@ -0,0 +1,110 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.kayenta.sql.migration;
+
+import com.netflix.kayenta.sql.config.SqlProperties;
+import com.netflix.kayenta.storage.ObjectType;
+import com.netflix.kayenta.storage.StorageService;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Scheduled;
+
+@Slf4j
+@RequiredArgsConstructor
+public class StorageDataMigrator {
+
+  private final SqlProperties sqlProperties;
+  private final StorageService sourceStorageService;
+  private final StorageService targetStorageService;
+  private final ExecutorService executorService;
+
+  private void migrate(ObjectType objectType) {
+    log.info("Migrating {}", objectType);
+
+    var sourceAccountName = sqlProperties.getMigration().getSourceAccountName();
+    var targetAccountName = sqlProperties.getMigration().getTargetAccountName();
+
+    var sourceObjectKeys =
+        sourceStorageService.listObjectKeys(sourceAccountName, objectType).stream()
+            .map(sourceObjectKey -> (String) sourceObjectKey.get("id"))
+            .collect(Collectors.toList());
+
+    var targetObjectKeys =
+        targetStorageService.listObjectKeys(targetAccountName, objectType).stream()
+            .map(sourceObjectKey -> (String) sourceObjectKey.get("id"))
+            .collect(Collectors.toList());
+
+    var objectKeysToMigrate =
+        sourceObjectKeys.stream()
+            .filter(
+                sourceObjectKey ->
+                    targetObjectKeys.stream()
+                        .filter(targetObjectKey -> Objects.equals(targetObjectKey, sourceObjectKey))
+                        .findFirst()
+                        .map(
+                            object -> {
+                              log.warn(
+                                  "Object for objectType: {}, key: {} already exists",
+                                  objectType,
+                                  sourceObjectKey);
+                              return object;
+                            })
+                        .isEmpty())
+            .collect(Collectors.toList());
+
+    if (objectKeysToMigrate.isEmpty()) {
+      log.info(
+          "No objects to migrate for objectType: {}, sourceObjectCount: {}, targetObjectCount: {}",
+          objectType,
+          sourceObjectKeys.size(),
+          targetObjectKeys.size());
+      return;
+    }
+
+    for (var objectKey : objectKeysToMigrate) {
+      executorService.submit(
+          () -> {
+            try {
+              var object =
+                  sourceStorageService.loadObject(sourceAccountName, objectType, objectKey);
+              targetStorageService.storeObject(targetAccountName, objectType, objectKey, object);
```

Comment:
Given this re-runs regularly, I'd expect this to output a LOT of data after a few cycles.  I'm wondering if there's a way to compare time stamps or something else so we don't spam logs once this is enabled.  There could be a lot of data here.